### PR TITLE
Update compiler options and msbuild options for 1.8

### PR DESCRIPTION
**Pull request by paulvanbrenk**

### CODE REVIEW

**Comment by sandersn** [comment]
:+1:

File: Compiler-Options.md [hunk 1]
```diff
@@ -27,16 +29,26 @@ Option | Shorthand | Description
...
 `--stripInternal`<sup>[1]</sup> | | Do not emit declarations for code that has an `/** @internal */` JSDoc annotation.
+`--suppressExcessPropertyErrors`<sup>[1]</sup> | | Suppress excess property checks for object literals.
 `--suppressImplicitAnyIndexErrors` | | Suppress `--noImplicitAny` errors for indexing objects lacking index signatures. See [issue #1232](https://github.com/Microsoft/TypeScript/issues/1232#issuecomment-64510362) for more details.
+`--allowUnusedLabels` | | Do not report errors on unused labels.
+`--noImplicitReturns` | | Report error when not all code paths in function return a value.
+`--noFallthroughCasesInSwitch` | | Report errors for fallthrough cases in switch statement.
+`--allowUnreachableCode` | | Do not report errors on unreachable code.
+`--forceConsistentCasingInFileNames` | | Disallow inconsistently-cased references to the same file.
+`--allowSyntheticDefaultImports` | | Allow default imports from modules with no default export. This does not affect code emit, just typechecking.
+`--allowJs` | | Allow javascript files to be compiled.
```
Comment (DanielRosenwasser) [inline Compiler-Options.md]: JavaScript

File: Compiler-Options.md [hunk 2]
```diff
@@ -12,14 +12,16 @@ Option | Shorthand | Description
 `--diagnostics` | | Show diagnostic information.
 `--emitBOM` | | Emit a UTF-8 Byte Order Mark (BOM) in the beginning of output files.
 `--emitDecoratorMetadata`<sup>[1]</sup> | | Emit design-type metadata for decorated declarations in source. See [issue #2577](https://github.com/Microsoft/TypeScript/issues/2577) for details.
+`--experimentalDecorators` | | Enables experimental support for ES7 decorators.
 `--inlineSourceMap` | | Emit a single file with source maps instead of having a separate file.
 `--inlineSources` | | Emit the source alongside the sourcemaps within a single file; requires `--inlineSourceMap` to be set.
 `--isolatedModules` | | Unconditionally emit imports for unresolved files.
 `--jsx` | | Support JSX in '.tsx' files: 'React' or 'Preserve'. See [[JSX]].
+`--reactNamespace` | | Specifies the object invoked for createElement and __spread when targeting 'react' JSX emit.
```
Comment (DanielRosenwasser) [inline Compiler-Options.md]: Use backticks around `createElement` and `__spread`

File: Setting-Compiler-Options-in-MSBuild-projects.md [hunk 3]
```diff
@@ -17,46 +17,54 @@ Compiler options can be specified using MSBuild properties within an MSBuild pro
...
+`--noFallthroughCasesInSwitch`<sup>[1]</sup>|TypeScriptNoFallthroughCasesInSwitch| boolean
+`--allowUnreachableCode`<sup>[1]</sup> |TypeScriptAllowUnreachableCode| boolean
+`--forceConsistentCasingInFileNames`<sup>[1]</sup>|TypeScriptForceConsistentCasingInFileNames| boolean
+`--allowSyntheticDefaultImports`<sup>[1]</sup>|TypeScriptAllowSyntheticDefaultImports| boolean
+`--project` | *Not supported in VS* |
+`--watch` | *Not supported in VS* |
+`--diagnostics` | *Not supported in VS* |
+`--listFiles` | *Not supported in VS* |
+`--noEmit` | *Not supported in VS* |
+*VS only option* | TypeScriptAdditionalFlags | *Any compiler option*
```
Comment (DanielRosenwasser) [inline Setting-Compiler-Options-in-MSBuild-projects.md]: Why???

File: Setting-Compiler-Options-in-MSBuild-projects.md [hunk 4]
```diff
@@ -17,46 +17,54 @@ Compiler options can be specified using MSBuild properties within an MSBuild pro
...
-`--suppressExcessPropertyErrors`             |  TypeScriptSuppressExcessPropertyErrors    | boolean
-`--moduleResolution`<sup>[1]</sup>           | TypeScriptModuleResolution                 | `Classic` or `Node`
-`--experimentalAsyncFunctions`<sup>[1]</sup> | TypeScriptExperimentalAsyncFunctions       | boolean
-`--jsx`<sup>[1]</sup>                        | TypeScriptJSXEmit                          | `React` or `Preserve`
-`--project`                                  | *Not supported in VS*                      |
-`--watch`                                    | *Not supported in VS*                      |
-`--diagnostics`                              | *Not supported in VS*                      |
-`--listFiles`                                | *Not supported in VS*                      |
-`--noEmit`                                   | *Not supported in VS*                      |
-*VS only option*                             | TypeScriptAdditionalFlags                  | *Any compiler option*
```
Comment (DanielRosenwasser) [inline Setting-Compiler-Options-in-MSBuild-projects.md]: Why???

Comment (paulvanbrenk) [inline Setting-Compiler-Options-in-MSBuild-projects.md]: Now it actually fitted on my screen while editing.

Comment (DanielRosenwasser) [inline Setting-Compiler-Options-in-MSBuild-projects.md]: How small is your screen?

Comment (sandersn) [inline Setting-Compiler-Options-in-MSBuild-projects.md]: Mine is 1080 pixels wide, so thinner is nicer.

**Pull request merged**